_blog/2018/egress-tcp.md

```diff
@@ -0,0 +1,306 @@
+---
+title: "Consuming External TCP Services"
+overview: Describes a simple scenario based on Istio Bookinfo sample
+publish_date: February 6, 2018
+subtitle: Egress rules for TCP traffic
+attribution: Vadim Eisenberg
+
+order: 92
+
+layout: blog
+type: markdown
+redirect_from: "/blog/egress-tcp.html"
+---
+{% include home.html %}
+
+In my previous blog post, [Consuming External Web Services]({{home}}/blog/2018/egress-https.html), I described how external services can be consumed by in-mesh Istio applications via HTTPS. In this post, I demonstrate consuming external services over TCP. I use the [Istio Bookinfo Sample Application]({{home}}/docs/guides/bookinfo.html), the version in which the book ratings data is persisted in a MySQL database. I deploy this database outside the cluster and will configure the _ratings_ microservice to use it. I define an [egress rule]({{home}}/docs/reference/config/istio.routing.v1alpha1.html#EgressRule) to allow the in-mesh applications access the external database.
+
+## Bookinfo Sample Application with External Ratings Database
+First, I set up a MySQL database instance to hold book ratings data, outside my Kubernetes cluster. Then I modify the [Bookinfo Sample Application]({{home}}/docs/guides/bookinfo.html) to use my database.
+
+### Setting up the Database for Ratings Data
```

Comment:
For all the headers in this post, and in your previous post, please use initial caps only, which is our convention:

## Setting up the database for ratings data

Reply:
@geeknoid thank you for the review!
A question - when citing another blog post on istio.io or a task, should I use the capital letters, as in the title of the task, or lower case letters? For example, should it be:

`the  [Consuming External Web Services](https://istio.io/blog/2018/egress-https.html) blog post`
the  [Consuming External Web Services](https://istio.io/blog/2018/egress-https.html) blog post

or 

`the  [consuming external web services](https://istio.io/blog/2018/egress-https.html) blog post`
the  [consuming external web services](https://istio.io/blog/2018/egress-https.html) blog post

Reply:
You should use the capitalized form that matches the title of the target.

Reply:
Thanks, it clear now.